exploration/text-vs-code.md

```diff
@@ -17,49 +19,361 @@ Status: **Proposed**
+		> It should be easy to do simple things; possible to do complex things; and impossible, or at least difficult, to do wrong things.
+		>
+		> —<a href="https://www.infoq.com/articles/API-Design-Joshua-Bloch/">Joshua Bloch, 2008</a>, author of <i>Effective Java</i>, etc.
+* Reducing the number of characters typed to the point it reduces clarity
+    - Beyond a certain point, there becomes a tradeoff between clarity and concision. (ex: Perl)
+
+## Constraints
+
+Some of the alternatives will require changes to the syntax to produce better usability.
+Similarly the current syntax could benefit from improvements if we decide to keep it.
+
+There are a limited number of sigils available for quoting.
+
+## Simple Messages
+
+In the Subcommittee meetings following Github discussions on Issues #493 and #499,
+the general consensus that formed for simple messages
+is that we allow them to be unquoted.
+
+("Simple messages" refers to messages consisting solely of a pattern, and thus are not complex messages.)
+
+Because the simple message pattern consists of the entire message,
+the pattern includes any leading or trailing whitespace.
+
+Given simple messages already being decided at a high level,
+the design decisions below for the proposed and alternative designs pertain specifically to complex messages.
+
+## Proposed Design
+
+Currently the syntax uses the first alternative below.
+
+## Alternatives Considered
+
+There are five candidates for handling the boundaries between code and patterns:
+
+1. Always quote non-simple patterns (current design)
+2. Never quote patterns (all whitespace is significant)
+3. Permit non-simple patterns to be quoted and trim unquoted whitespace
+4. Trim all unquoted whitespace, but do not permit quoting non-simple patterns
+5. Selectively trim patterns (all whitespace is otherwise significant)
+
+### Always Quote
+
+Pros:
+- The boundary between pattern and code is always clear.
+- The quoting reduces the number of in-pattern escapes to the open/close sequence.
+  and the placeholder sequence sigils.
+- Since the pattern is already quoted, translators never have to add pattern quotes
+  in order to add PEWs to a given pattern.
+  This also might avoid some tools forcing escaping on added quotes that are needed.
+
+Cons:
+- Requires matching open/close quotes.
+
+### Never Quote Patterns
+
+In this alternative, all non-code whitespace is significant. 
+We have to use a slightly different syntax in the example, so that
+the boundary between code and pattern works.
+>```
+>{{
+>   match {$var}
+>   {when *} This pattern has a space in front (it's between \} and This)
+>   {when other}
+>      This pattern has a newline and six spaces in front of it
+>   {when moo}This pattern has no spaces in front of it, but an invisible space at the end
+>}}
+>```
+
+Pros:
+- WYSIWYG (on steroids)
+
+Cons:
+- Probably not a serious alternative: the example
+  includes any number of obvious footguns that have to be addressed
```

Comment:
This seems really rather opinionated. In many ways, this is the same as the "Always quote" solution, except that the pattern delimiters are `}…{` instead of `{{…}}`. So the unnamed footguns probably apply to that alternative as well.

Reply:
It is opinionated. When I wrote it, I was being lazy by not enumerating the issues.

> this is the same as the "Always quote" solution, except that the pattern delimiters are }…{ instead of {{…}}. So the unnamed footguns probably apply to that alternative as well.

No, this is incorrect.

`{{...}}` encloses all and only the whitespace that is intentional in the pattern, with `{{` and `}}` forming the pattern boundary. These boundary characters are visible.

`}...{` makes all whitespace in the variant block meaningful. It effectively prohibits a multiline representation of a message, because the newlines are always meaningful. It also means that trailing spaces (which are invisible) have meaning. 

To make a message multiline, you have to put the whitespace inside the key.:

```
{{
match {$var}
{when 0
}This has no newline or space.{
when one}
This has a newline at the start.{
when *} This has a space at the start and six spaces and a newline at the end.      
}}
```

Reply:
If we take your example above, and after the `match` replace each `}` with `{{` and each `{` with `}}`, we get this:

```
{{
match {$var}
}}when 0
{{This has no newline or space.}}
when one{{
This has a newline at the start.}}
when *{{ This has a space at the start and six spaces and a newline at the end.      
}}
```

Ignoring the specifics of what's happening with the preamble, that seems pretty similar to me. It's just that we're conditioned to look at the `{` and `}` a certain way.